docs/source/onnx.rst

```diff
@@ -249,6 +249,31 @@ E.g.: ::
     out = model(*inputs)
     torch.onnx.export(model, inputs, 'loop_and_list.onnx', opset_version=11, example_outputs=out)
 
+Write pytorch model in Torch way
+--------------------------------
+
+Pytorch models can be written using numpy manipulations, but this is not proper when we convert to the onnx model.
+For the trace-based exporter, tracing treats the numpy values as the constant node,
+therefore it calculates the wrong result if we change the input.
+So the pytorch model need implement using torch operators.
+For example, do not use numpy operators on numpy tensors: ::
+
+    np.concatenate((x, y, z), axis=1)
+
+do not convert to numpy types: ::
+
+    y = x.astype(np.int)
+
+Always use torch tensors and torch operators: torch.concat, etc.
+In addition, Dropout layer need defined in init function so that inferencing can handle it properly, i.e., ::
```

Comment:
Does this only apply to Dropout, what about Batchnorm, or any other op that operates differently in training vs inferencing?

Reply:
Not sure about Batchnorm. The difference is that Dropout in inferencing will be Identity, but Batchnorm is Batchnorm for both training/inferencing case. This may cause the difference. So far we only meet the Dropout layer issue, so we just include Dropout here. We can add more if we further meet new issues.